force-app/main/default/classes/HOT_MessagesNotification.cls

```diff
@@ -0,0 +1,45 @@
+public with sharing class HOT_MessagesNotification {
+    public HOT_MessagesNotification() {
+        
+    }
+    public static void Notification(List<Id> threadIds){
+        List<Thread__c> threads=[SELECT Id, HOT_Dispatcher__c, HOT_Subject__c FROM Thread__c WHERE Id IN:threadIds];
+
+        CustomNotificationType notificationType = [
+            SELECT Id, DeveloperName
+            FROM CustomNotificationType
+            WHERE DeveloperName = 'HOT_NotifyDispatcher'
+        ];
+
+        for(Thread__c thread: threads){
+            Messaging.CustomNotification notification = new Messaging.CustomNotification();
+            notification.setTitle('Ny melding');
+            notification.setBody(thread.HOT_Subject__c);
+            notification.setNotificationTypeId(notificationType.Id);
+            notification.setTargetId(thread.Id);
+
+            Set<String> recipients=new Set<String>();
+            recipients.add(thread.HOT_Dispatcher__c);
+
+            if(thread.HOT_Dispatcher__c!=null){
+
+                HOT_NotificationHandler.sendNotification(notification, recipients, thread);
+                
+            }
+            if(thread.HOT_Dispatcher__c==null){
+                List<Id>ownerIds=new List<Id>();
+                ownerIds.add(thread.OwnerId);
+                Map<Id, ID> groupByQueue = HOT_NotificationHandler.getGroupIdByQueueId(ownerIds);
+
+
+                Set<String> recipientsqueue=new Set<String>();
+                recipientsqueue.add(thread.HOT_Dispatcher__c);
+                for(Id owner: groupByQueue.keySet()){
+                    recipientsqueue.add(owner);
+                    HOT_NotificationHandler.sendNotification(notification, recipientsqueue, thread);
```

Comment:
Kanskje ikke nødvendig med en for-loop her?
Kan vi rydde litt opp i koden, slik at logikken ligger i hva vi legger i recipients-lista, og vi kaller bare på sendNotification-metoden én gang?

getGroupIdByQueueId er en metode som gjør flere soql spørringer. Kallet på denne metoden kan ikke ligge i en for loop.